cmd/site-replication.go

```diff
@@ -3058,3 +3272,995 @@ func (c *SiteReplicationSys) PeerEditReq(ctx context.Context, arg madmin.PeerInf
+				logger.LogIf(ctx, fmt.Errorf("Error healing sse metadata from peer site %s : %w", latestPeerName, err))
+			}
+			continue
+		}
+
+		admClient, err := c.getAdminClient(ctx, dID)
+		if err != nil {
+			return wrapSRErr(err)
+		}
+		peerName := info.Sites[dID].Name
+		err = admClient.SRPeerReplicateBucketMeta(ctx, madmin.SRBucketMeta{
+			Type:      madmin.SRBucketMetaTypeSSEConfig,
+			Bucket:    bucket,
+			SSEConfig: latestSSEConfig,
+		})
+		if err != nil {
+			logger.LogIf(ctx, c.annotatePeerErr(peerName, "SRPeerReplicateBucketMeta", fmt.Errorf("Error healing SSE config metadata for peer %s from peer %s : %s", peerName, latestPeerName, err.Error())))
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *SiteReplicationSys) healOLockConfigMetadata(ctx context.Context, objAPI ObjectLayer, bucket string, info srStatusInfo) error {
+	c.RLock()
+	defer c.RUnlock()
+	if !c.enabled {
+		return nil
+	}
+	var (
+		latestID, latestPeerName string
+		lastUpdate               time.Time
+		latestObjLockConfig      *string
+	)
+
+	bs, ok := info.BucketStats[bucket]
+	if !ok {
+		return nil
+	}
+	for dID, ss := range bs {
+		if lastUpdate.IsZero() {
+			lastUpdate = ss.meta.ObjectLockConfigUpdatedAt
+			latestID = dID
+			latestObjLockConfig = ss.meta.ObjectLockConfig
+		}
+		// avoid considering just created buckets as latest. Perhaps this site
+		// just joined cluster replication and yet to be sync'd
+		if ss.meta.CreatedAt.Equal(ss.meta.ObjectLockConfigUpdatedAt) {
+			continue
+		}
+		if ss.meta.ObjectLockConfig != nil && ss.meta.ObjectLockConfigUpdatedAt.After(lastUpdate) {
+			lastUpdate = ss.meta.ObjectLockConfigUpdatedAt
+			latestID = dID
+			latestObjLockConfig = ss.meta.ObjectLockConfig
+		}
+	}
+	latestPeerName = info.Sites[latestID].Name
+	var latestObjLockConfigBytes []byte
+	if latestObjLockConfig != nil {
+		latestObjLockConfigBytes = []byte(*latestObjLockConfig)
+	}
+
+	for dID, bStatus := range bs {
+		if !bStatus.OLockConfigMismatch {
+			continue
+		}
+		if isBucketMetadataEqual(latestObjLockConfig, bStatus.meta.ObjectLockConfig) {
+			continue
+		}
+		if dID == globalDeploymentID {
+			if err := globalBucketMetadataSys.Update(ctx, bucket, objectLockConfig, latestObjLockConfigBytes); err != nil {
+				logger.LogIf(ctx, fmt.Errorf("Error healing objectlock config metadata from peer site %s : %w", latestPeerName, err))
+			}
+			continue
+		}
+
+		admClient, err := c.getAdminClient(ctx, dID)
+		if err != nil {
+			return wrapSRErr(err)
+		}
+		peerName := info.Sites[dID].Name
+		err = admClient.SRPeerReplicateBucketMeta(ctx, madmin.SRBucketMeta{
+			Type:   madmin.SRBucketMetaTypeObjectLockConfig,
+			Bucket: bucket,
+			Tags:   latestObjLockConfig,
+		})
+		if err != nil {
+			logger.LogIf(ctx, c.annotatePeerErr(peerName, "SRPeerReplicateBucketMeta", fmt.Errorf("Error healing object lock config metadata for peer %s from peer %s : %w", peerName, latestPeerName, err)))
+		}
+		return err
```

Comment:
```suggestion
		if err != nil {
			logger.LogIf(ctx, c.annotatePeerErr(peerName, "SRPeerReplicateBucketMeta", fmt.Errorf("Error healing object lock config metadata for peer %s from peer %s : %w", peerName, latestPeerName, err)))
			return err
		}
```

This can return to nil - but in general in healing I think we should not quit for the first error that we see (for example if one site is offline, then we will never able to heal other sites). We can do that in a subsequent PR if we agree.

Reply:
yes, more improvements are needed.